torch/distributed/nn/functional.py

```diff
@@ -0,0 +1,263 @@
+import torch
+from torch.autograd import Function
+import torch.distributed as dist
+
+
+def broadcast(tensor, src, group=dist.group.WORLD):
+    """
+    Broadcasts the tensor to the whole group.
+
+    ``tensor`` must have the same number of elements in all processes
+    participating in the collective.
+
+    Arguments:
+        tensor (Tensor): Data to be sent if ``src`` is the rank of current
+            process, and tensor to be used to save received data otherwise.
```

Comment:
`tensor` is not actually used to save the received data here. A new Tensor is created to contain it. 
We should either update the doc or make the function actually inplace no?

Reply:
let's modify the docs